Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -4413,39 +4521,150 @@ def panorama_query_logs_command(args: dict):
     rule = args.get('rule')
     filedigest = args.get('filedigest')
     url = args.get('url')
+    use_polling = args.get('polling', 'false') == 'true'
+    job_id = args.get('job_id')
+    cmd = demisto.command()
+    interval_in_seconds = int(args.get('interval_in_seconds', 60))
+    timeout = int(args.get('timeout', 600))
+    script_results = []
 
     if query and (address_src or address_dst or zone_src or zone_dst
                   or time_generated or action or port_dst or rule or url or filedigest):
         raise Exception('Use the free query argument or the fixed search parameters arguments to build your query.')
 
-    result = panorama_query_logs(log_type, number_of_logs, query, address_src, address_dst, ip_,
-                                 zone_src, zone_dst, time_generated, action,
-                                 port_dst, rule, url, filedigest)
+    if use_polling:
+        ScheduledCommand.raise_error_if_not_supported()
+        if not job_id:
+
+            # create new search
+            result = panorama_query_logs(log_type, number_of_logs, query, address_src, address_dst, ip_,
+                                         zone_src, zone_dst, time_generated, action,
+                                         port_dst, rule, url, filedigest)
+
+            if result['response']['@status'] == 'error':
```

Comment:
like above, i would extract the ```result['response']``` into a variable and replace it in every place it being used in the function. the same for every other dict value which is being taken more than once.